app.js

```diff
@@ -0,0 +1,195 @@
+};
+
+function instantiateLocations() {
+  for(var i = 0; i < breakfastLocationData.length; i++) {
+    new Location(breakfastLocationData[i]);
+    new Location(lunchLocationData[i]);
+    new Location(dinnerLocationData[i]);
+  }
+};
+
+instantiateLocations();
+
+var map;
+
+function googleMap() {
+  map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: 47.6050, lng: -122.3344},
+    zoom: 13
+  });
+};
+
+// start filtering meal type locations IF CLICKED. saving for now just in case
+// var breakfast = document.getElementById("breakfast");
+// var lunch = document.getElementById("lunch");
+// var dinner = document.getElementById("dinner");
+
+//display all Breakfast locations when clicked
+// function clickBreakfast(event) {
+//   for(var i = 0; i < breakfastLocationData.length; i++) {
+//     alert(breakfastLocationData[i].name);
+//   }
+// }
+// breakfast.addEventListener("click", clickBreakfast);
+// ^ end filtering click event for meal types ^
+
+//helper function to create table elements
+function makeAnElementWithText(element, textContent, parent) {
+  var childEl = document.createElement(element);
+  childEl.textContent = textContent;
+  parent.appendChild(childEl);
+};
+
+function setClassOfAddressCells(element, textContent, parent) {
+  var childEl = document.createElement(element);
+  childEl.setAttribute('class', 'address');
+  childEl.textContent = textContent;
+  parent.appendChild(childEl);
+};
+
+//helper function to create reservation button
+function makeAReservationButton(idName, parent) {
+  var buttonEl = document.createElement('button');
+  buttonEl.setAttribute('id', idName);
+  buttonEl.textContent = 'Make a Reservation';
+  parent.appendChild(buttonEl);
+};
+
+//helper function to create table rows
+function createRow(idName, rowElement, El, tC1, tC2, tC3, tC4, tC5) {
+  var tableEl = document.getElementById(idName);
+  var rowEl = document.createElement(rowElement);
+  makeAnElementWithText(El, tC1, rowEl);
+  makeAnElementWithText(El, tC2, rowEl);
+  setClassOfAddressCells(El, tC3, rowEl);
+  makeAnElementWithText(El, tC4, rowEl);
+  makeAnElementWithText(El, tC5, rowEl);
+  makeAReservationButton(tC1, rowEl);
+  tableEl.appendChild(rowEl);
+};
+
+//loop to create table with object instances in allLocations array
+function populateTable() {
+  var tableEl = document.getElementById('foodTable');
+  tableEl.innerHTML = ' ';
+  var rowEl = document.createElement('thead');
+  makeAnElementWithText('th', 'Location Name', rowEl);
+  makeAnElementWithText('th', 'Neighborhood', rowEl);
+  makeAnElementWithText('th', 'Address', rowEl);
+  makeAnElementWithText('th', 'Requirements', rowEl);
+  makeAnElementWithText('th', 'Meal', rowEl);
+  tableEl.appendChild(rowEl);
+  for (var i = 0; i < allLocations.length; i++) {
+    createRow('foodTable', 'tr', 'td', allLocations[i].name, allLocations[i].hood, allLocations[i].address, allLocations[i].restrictions, allLocations[i].mealType);
```

Comment:
so many arguments! why not `createRow('foodTable', 'tr', 'td', allLocations[i]` here and above, 
```function createRow(idName, rowElement, El, location) {
...
makeAnElementWithText(El, tC1, location.name);
...```